lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -1642,14 +1642,17 @@ void TFDeabstraction::checkAttributesAndFormGraphOps() {
   SmallVector<SILValue, 32> valuesToCheck;
 
   for (auto *op : tensorOps) {
-    for (auto &operand : op->getAllOperands()) {
+    auto opInfo = SILTensorOpInfo::decode(op);
```

Comment:
Should we assert that opInfo is valid?

Reply:
Thank you for the review, Gogul!

I tried asserting `opInfo` to be non-null here, however it breaks a number of tests... So I think we'd better go with the more robust way (i.e., testing its validity at [this line](https://github.com/apple/swift/pull/18249/files/054e230c322f2ef447b8db3161e49da0507da598#diff-d86f33ececb50054de2a79dcc8a95642R1652)).

Reply:
Thanks, Ding. In that case, let us leave it as is for this PR. 

However, isn't the failing tests an indication that we have collecting non tensor operations in tensorOps? Should we file a bug?

Reply:
Christ advised a different fix where `opInfo` is not involved. Anyway, I can take a look at this issue later.